tests/drivers/gpio/gpio_basic_api/src/test_gpio_port.c

```diff
@@ -81,11 +81,39 @@ static int setup(void)
 	zassert_equal(rc, 0,
 		      "pin config input failed");
 
+	if (IS_ENABLED(CONFIG_GPIO_GET_DIRECTION)) {
```

Comment:
Perhaps move these into their own test case instead of piggybacking on setup - and properly indicate if they are not tested by skipping the test?

Reply:
I see you have moved this to a completely new test suite? That's perhaps overkill. I merely suggested a new, dedicated test case (function) in the `gpio_basic_api` test suite. Either way works, though.